File: app/src/main/java/com/dominikgaller/alpinebooking/shared/domain/event/DomainEvent.java
package com.dominikgaller.alpinebooking.shared.domain.event;

/**
 * Marker interface for all domain events across all bounded contexts.
 *
 * <p>Implementing types are immutable records that describe facts
 * which have occurred in the domain (named in past tense).
 *
 * <p>Framework-free: no Spring, no IO.
 *
 * <p>SDD: See {@code documentation/architecture.definition.md}, section 9 (Shared Kernel).
 */
public interface DomainEvent {
}

File: src/main/kotlin/com/example/service/shared/domain/event/DomainEvent.kt
package com.example.service.shared.domain.event

/**
 * Marker interface for all domain events across all bounded contexts.
 *
 * Implementing types are immutable data classes describing facts that have occurred in
 * the domain, named in the past tense.
 *
 * Framework-free: no Spring, no IO.
 *
 * SDD: see `documentation/architecture.definition.md`, section 9 (Shared Kernel).
 */
interface DomainEvent
